lib/posthog/client.rb

```diff
@@ -342,10 +346,31 @@ def shutdown
 
     private
 
+    # before_send should run immediately before the event is sent to the queue.
+    # @param [Object] action The event to be sent to PostHog
+    # @return [null, Object, nil] The processed event or nil if the event should not be sent
+    def process_before_send(action)
+      return if action.nil? || action.empty?
```

Comment:
We're always returning `nil` even in the case where it's simply empty. Maybe should be
```suggestion
      return action if action.nil? || action.empty?
```